### PR TITLE
Adds check for duplicate unit key

### PR DESCRIPTION
**Pull request by dkliban**
re #1406

### CODE REVIEW

**Comment by dkliban** [comment]
ok test

File: pulp_puppet_plugins/pulp_puppet/plugins/importers/directory.py [hunk 1]
```diff
@@ -265,7 +267,10 @@ def _import_modules(self, module_paths):
             _logger.debug(IMPORT_MODULE, dict(mod=module_path))
 
             module.set_storage_path(os.path.basename(module_path))
-            module.save_and_import_content(module_path)
+            try:
+                module.save_and_import_content(module_path)
+            except NotUniqueError:
+                module = module.__class__.objects.filter(**module.unit_key).first()
```
Comment (rbarlow) [inline pulp_puppet_plugins/pulp_puppet/plugins/importers/directory.py]: You can use .get() as discussed on the other similar PRs.

File: pulp_puppet_plugins/pulp_puppet/plugins/importers/upload.py [hunk 2]
```diff
@@ -45,8 +47,10 @@ def handle_uploaded_unit(repo, type_id, unit_key, metadata, file_path, conduit):
 
     uploaded_module = Module.from_metadata(extracted_data)
     uploaded_module.set_storage_path(os.path.basename(new_file_path))
-    uploaded_module.save_and_import_content(new_file_path)
-
+    try:
+        uploaded_module.save_and_import_content(new_file_path)
+    except NotUniqueError:
+                uploaded_module = uploaded_module.__class__.objects.filter(**uploaded_module.unit_key).first()
```
Comment (rbarlow) [inline pulp_puppet_plugins/pulp_puppet/plugins/importers/upload.py]: The indentation here is weird - you should left shift it so it's only 4 spaces.

**Comment by rbarlow** [comment]
Definitely fix the indentation before merging, and again consider .get(). Either write tests (unit or smash, your choice), or at least file a pulp-smash about it.

**Pull request merged**